src/util-base64.c

```diff
@@ -93,68 +94,136 @@ static inline void DecodeBase64Block(uint8_t ascii[ASCII_BLOCK], uint8_t b64[B64
     ascii[2] = (uint8_t) (b64[2] << 6) | (b64[3]);
 }
 
+static inline bool AmpleDestinationSize(
+        uint32_t dest_size, uint32_t total_decoded, uint32_t cur_decoded)
+{
+    if (dest_size < total_decoded + cur_decoded) {
+        return false;
+    }
+    return true;
+}
+
 /**
- * \brief Decodes a base64-encoded string buffer into an ascii-encoded byte buffer
+ * \brief Decode a base64 encoded string as per RFC 2045.
+ *        RFC 2045 states that any characters that do not fall under the Base64
+ *        alphabet must be skipped by the decoding software.
+ *        Following are some important considerations:
+ *        1. This Decoding algorithm is used by MIME parser currently.
+ *        2. The number of decoded bytes are constrained by the destination buffer size.
+ *        3. The leftover bytes are not handled by the decoder but the caller.
  *
- * \param dest The destination byte buffer
- * \param dest_size The destination byte buffer size
- * \param src The source string
- * \param len The length of the source string
- * \param consumed_bytes The bytes that were actually processed/consumed
- * \param decoded_bytes The bytes that were decoded
- * \param mode The mode in which decoding should happen
+ * \param dest destination buffer
+ * \param dest_size destination buffer size
+ * \param src base64 encoded string
+ * \param len length of the base64 encoded string
+ * \param consumed_bytes number of bytes successfully consumed by the decoder
+ * \param decoded_bytes number of bytes successfully decoded by the decoder
  *
- * \return Error code indicating success or failures with parsing
+ * \return Base64Ecode BASE64_ECODE_OK if all went well
+ *                     BASE64_ECODE_BUF if destination buffer got full before all could be decoded
  */
-Base64Ecode DecodeBase64(uint8_t *dest, uint32_t dest_size, const uint8_t *src, uint32_t len,
-        uint32_t *consumed_bytes, uint32_t *decoded_bytes, Base64Mode mode)
+static inline Base64Ecode DecodeBase64RFC2045(uint8_t *dest, uint32_t dest_size, const uint8_t *src,
```

Comment:
I am not sure about the philosophy of the commit as it seems to bring in more complexity... (multiply by 3 the lines of code to review...)

Reply:
ok. Have you tried opening the changes in place in the file? diff is messy indeed. Lmk if you'd still like to reject this. Thank you!

Reply:
Also to note: I added lots of fn docs, that may be adding to the increase in the number of lines.